tensorflow/compiler/xla/service/gpu/cudnn_batchnorm_rewriter.cc

```diff
@@ -143,17 +185,33 @@ Status Visitor::HandleBatchNormTraining(HloInstruction* batch_norm) {
           computation_->AddInstruction(HloInstruction::CreateBroadcast(
               variance_plus_epsilon->shape(), epsilon, {}))));
 
-  // Repackage the results.
-  std::unique_ptr<HloInstruction> new_tuple = HloInstruction::CreateTuple({
-      computation_->AddInstruction(HloInstruction::CreateGetTupleElement(
-          libcall->shape().tuple_shapes(0), libcall, 0)),
+  HloInstruction* new_tuple =
+      computation_->AddInstruction(HloInstruction::CreateTuple({
+          computation_->AddInstruction(HloInstruction::CreateGetTupleElement(
+              libcall->shape().tuple_shapes(0), libcall, 0)),
+          computation_->AddInstruction(HloInstruction::CreateGetTupleElement(
+              libcall->shape().tuple_shapes(1), libcall, 1)),
+          variance,
+      }));
+
+  HloInstruction* new_gte =
       computation_->AddInstruction(HloInstruction::CreateGetTupleElement(
-          libcall->shape().tuple_shapes(1), libcall, 1)),
-      variance,
-  });
+          new_tuple->shape().tuple_shapes(0), new_tuple, 0));
+
+  if (IsF32BatchNormWithFP16Inputs(batch_norm)) {
+    new_gte = AddConvert(new_gte, F32);
+  }
+  // Repackage the results. Athough this tuple is redundant when convert is not
+  // inserted, TupleSimplifier eliminates the Tuple eventually
+  std::unique_ptr<HloInstruction> replacing_tuple = HloInstruction::CreateTuple(
+      {new_gte,
+       computation_->AddInstruction(HloInstruction::CreateGetTupleElement(
+           new_tuple->shape().tuple_shapes(1), new_tuple, 1)),
+       computation_->AddInstruction(HloInstruction::CreateGetTupleElement(
+           new_tuple->shape().tuple_shapes(1), new_tuple, 2))});
```

Comment:
Should this be `new_tuple->shape().tuple_shapes(2)`?

Reply:
Correct..Sorry that is a bug